src/main/java/org/jboss/logmanager/BootstrapConfiguration.java

```diff
@@ -0,0 +1,444 @@
+
+    /**
+     * Returns the level used for the bootstrapped messages.
+     *
+     * @return the level to use
+     */
+    public Level getLevel() {
+        return level == null ? BOOTSTRAP_LEVEL : level;
+    }
+
+    /**
+     * Allows the default level of {@link Level#INFO} to be overridden. This can also be controlled by the
+     * {@code org.jboss.logmanager.bootstrap.level} system property.
+     * <p>
+     * Note that the level used here will be set on the root logger. If a different root logging level is desired it
+     * must be explicitly set before the {@link LogContext#configurationComplete()} is invoked.
+     * </p>
+     *
+     * @param level the log level to use for bootstrapped messages or {@code null} to use the
+     *              {@code org.jboss.logmanager.bootstrap.level} system property.
+     *
+     * @return this configuration
+     */
+    public BootstrapConfiguration setLogLevel(final Level level) {
+        this.level = level;
+        return this;
+    }
+
+    /**
+     * Returns the maximum size of the queue for the bootstrapped log messages.
+     *
+     * @return the maximum queue size
+     */
+    public int getQueueSize() {
+        return queueSize;
+    }
+
+    /**
+     * Sets the maximum size of the queue for bootstrapped log messages. If the size is less than or equal to 0 the
+     * system property {@code org.jboss.logmanager.bootstrap.queue-size} will be used to determine the size. If the
+     * system property is not set 10,000 will be used.
+     *
+     * @param queueSize the queue size
+     *
+     * @return this configuration
+     */
+    public BootstrapConfiguration setQueueSize(final int queueSize) {
+        if (queueSize <= 0) {
+            this.queueSize = BOOTSTRAP_QUEUE_SIZE;
+        } else {
+            this.queueSize = queueSize;
+        }
+        return this;
+    }
+
+    /**
+     * If {@linkplain #isBootstrapEnabled() enabled} this will return a configured {@link Bootstrap}, otherwise
+     * {@code null} will be returned.
+     *
+     * @param loggerNode the logger node to associate with the bootstrap loggers
+     *
+     * @return a configured bootstrap or {@code null} if bootstrapping isn't enabled
+     */
+    Bootstrap build(final LoggerNode loggerNode) {
+        if (enabled) {
+            return new Bootstrap(loggerNode, this);
+        }
+        return null;
+    }
+
+    private Supplier<Handler> getDefaultHandler() {
+        if ("console".equalsIgnoreCase(BOOTSTRAP_FAILED_LOG_TYPE)) {
+            return new Supplier<Handler>() {
+                @Override
+                public Handler get() {
+                    return new ConsoleHandler(new PatternFormatter(PATTERN));
+                }
+            };
+        }
+        return new Supplier<Handler>() {
+            @Override
+            public Handler get() {
+                try {
+                    return new FileHandler(new PatternFormatter(PATTERN), BOOTSTRAP_FAILED_LOG_FILE);
```

Comment:
We probably should create this with `append` being `true`. Different log contexts may overwrite each other.